plugins/hls-stan-plugin/src/Ide/Plugin/Stan.hs

```diff
@@ -0,0 +1,119 @@
+                                                 Pretty (..), Recorder,
+                                                 RuleResult, Rules,
+                                                 ShowDiagnostic (..),
+                                                 TypeCheck (..), WithPriority,
+                                                 action, cmapWithPrio, define,
+                                                 getFilesOfInterestUntracked,
+                                                 hscEnv, msrModSummary,
+                                                 tmrTypechecked, use, uses)
+import           Development.IDE.Core.RuleTypes (HieAstResult (..))
+import           Development.IDE.Core.Rules     (getSourceFileSource)
+import qualified Development.IDE.Core.Shake     as Shake
+import           Development.IDE.GHC.Compat     (HieASTs (HieASTs),
+                                                 RealSrcSpan (..), mkHieFile',
+                                                 mkRealSrcLoc, mkRealSrcSpan,
+                                                 runHsc, srcSpanEndCol,
+                                                 srcSpanEndLine,
+                                                 srcSpanStartCol,
+                                                 srcSpanStartLine, tcg_exports)
+import           Development.IDE.GHC.Error      (realSrcSpanToRange)
+import           GHC.Generics                   (Generic)
+import           HieTypes                       (HieASTs, HieFile)
+import           Ide.Types                      (PluginDescriptor (..),
+                                                 PluginId,
+                                                 defaultPluginDescriptor)
+import qualified Language.LSP.Types             as LSP
+import           Stan.Analysis                  (Analysis (..), runAnalysis)
+import           Stan.Core.Id                   (Id (..))
+import           Stan.Inspection                (Inspection (..))
+import           Stan.Inspection.All            (inspectionsIds, inspectionsMap)
+import           Stan.Observation               (Observation (..))
+
+descriptor :: Recorder (WithPriority Log) -> PluginId -> PluginDescriptor IdeState
+descriptor recorder plId = (defaultPluginDescriptor plId) {pluginRules = rules recorder}
+
+newtype Log = LogShake Shake.Log deriving (Show)
+
+instance Pretty Log where
+  pretty = \case
+    LogShake log -> pretty log
+
+data GetStanDiagnostics = GetStanDiagnostics
+  deriving (Eq, Show, Generic)
+
+instance Hashable GetStanDiagnostics
+
+instance NFData GetStanDiagnostics
+
+type instance RuleResult GetStanDiagnostics = ()
+
+rules :: Recorder (WithPriority Log) -> Rules ()
+rules recorder = do
+  define (cmapWithPrio LogShake recorder) $
+    \GetStanDiagnostics file -> do
+      hie <- fromJust <$> getHieFile file
+      let enabledInspections = HM.fromList [(LSP.fromNormalizedFilePath file, inspectionsIds)]
```

Comment:
Is this idea to make this configurable? Comment?

Reply:
We can enable Inspections by category: https://github.com/kowainik/stan/blob/main/src/Stan/Inspection/All.hs#L31.

Or we can add `excludeInspections` option.

Also Stan may look for cabal config to look for extensions. Is it possible to ask HLS for cabal file?

Reply:
Maybe a configure file like https://github.com/kowainik/stan/blob/main/.stan.toml?

Reply:
I'm not saying we should necessarily make it configurable now, but it would be good to write something down here about it.

e.g.
```
-- In the future we may want to control which inspections are enabled based on ???
```

> Also Stan may look for cabal config to look for extensions.

HLS knows the extensions enabled for a file directly, if `stan` will let you set them.

Reply:
> HLS knows the extensions enabled for a file directly

I guess Stan waits for cabal enabled extensions, not extensions enabled directly, cuz function used for collecting extensions is named `createCabalExtensionsMap`. https://github.com/kowainik/stan/blob/main/src/Stan.hs#L74

Reply:
Comment done.